docs/optimization-framework/ab-testing.md

```diff
@@ -85,30 +87,141 @@ Returns the A/B Test markup for client side processing.
 - `$default_content`: The default content for the control test.
 - `$args`: An optional array of data to pass through to `variant_callback`.
 
+### Example basic test
+
+An example for testing a CTA (call to action) button with different text would look like:
+
 ```php
+// feature.php
+
 namespace Altis\Analytics\Experiments;
 
-// Register the test.
-register_post_ab_test( 'featured_images', [
-	'rest_api_variants_type' => 'integer',
-	'goal' => 'click',
-	'variant_callback' => function ( $attachment_id, $post_id, $args ) {
-		return wp_get_attachment_image(
-			$attachment_id,
-			$args['size'],
-			false,
-			$args['attr']
-		);
-	}
-] );
+add_action( 'init', function() {
 
-// Apply the test by filtering some standard output.
-add_filter( 'post_thumbnail_html', function ( $html, $post_id, $post_thumbnail_id, $size, $attr ) {
-	return output_ab_test_html_for_post( 'featured_images', $post_id, $html, [
-		'size' => $size,
-		'attr' => $attr,
+	// Register the test.
+	register_post_ab_test( 'cta_button_text', [
+		'rest_api_variants_type' => 'string',
+		'goal' => 'click',
+		'variant_callback' => function ( $variant_value, $post_id, $some_extra_args ) {
+			return render_the_custom_cta_callback( $variant_value, $post_id, $some_extra_args );
+		},
 	] );
-}, 10, 5 );
+
+	// Apply the test by using `output_ab_test_html_for_post`.
+	add_action( 'the_custom_cta', function () {
+		$post_id = get_the_ID();
+
+		return output_ab_test_html_for_post( 'cta_button_text', $post_id, $some_extra_args );
+	}, 10, 5 );
+
+} );
+```
+
+### Example test with a custom JS control
+
+If the value of the variant is not strictly textual, a custom control might be needed to use within the Experiement Side Panel of the Editor, which will need to be registered in the following way:
+
+```php
+// feature.php
+
+add_action( 'init', function() {
+
+	// Register the test.
+	register_post_ab_test( 'cta_button_style', [
+		'rest_api_variants_type' => 'string',
+		'goal' => 'click',
+		'variant_callback' => function ( $variant_value, $post_id, $some_extra_args ) {
+			return render_the_custom_cta_callback( $variant_value, $post_id, $some_extra_args );
+		},
+		'editor_scripts' => [
+			'url/to/feature.js' => [
+				'wp-blocks',
+				'wp-i18n',
+				'wp-plugins',
+				'wp-editor',
+				'wp-components',
+				'wp-core-data',
+				'wp-edit-post',
+				'...',
+			],
+		],
+	] );
+
+	// Apply the test by using `output_ab_test_html_for_post`.
+	add_action( 'the_custom_cta', function () {
+		return output_ab_test_html_for_post( 'cta_button_style', get_the_ID(), $some_extra_args );
+	}, 10, 5 );
+
+} );
+```
+
+To register a custom editor input, and override some of the test options.
+
+```js
+// feature.js
+
+import { EditControl, PreviewControl } from 'some/custom/library';
```

Comment:
I don't really understand this JS example and I'm not confident that I could use it to create my own custom test as it is. Can we use a more complete real-world example? Maybe take a pared down version of the title test implementation that changes the output of some post meta or custom field?